tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -866,10 +866,11 @@ protected synchronized void handle(DAGAppMasterEvent event) {
       taskCommunicatorManager.dagComplete(cleanupEvent.getDag());
       nodes.dagComplete(cleanupEvent.getDag());
       containers.dagComplete(cleanupEvent.getDag());
-      TezTaskAttemptID.clearCache();
-      TezTaskID.clearCache();
-      TezVertexID.clearCache();
-      TezDAGID.clearCache();
+
+      // Hint to the JVM to clean up anything no longer needed; be ready for the next DAG
+      System.gc();
+      System.runFinalization();
+
```

Comment:
I'm not sure if it is helpful to role this change into this ticket or not.  However, the `clearCache` stuff is pretty useless.  All it does is make the map entries available for garbage collection,... which they already are by design.

However, this here will hint to the JDK to actually clean up that stuff.  Hinting to the JDK at this points seems like a reasonable thing to do between DAGs, but I'm not sure if it's worthwhile.